.github/workflows/main.yaml

```diff
@@ -0,0 +1,91 @@
+name: CI/CD
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

Comment:
[Open Discussion]Are we sure we don't want to run the CI on pr with base branch not set to `main`? E.g. I've opened a PR from branch `x` to `main` (and this trigger the CI), but then someone opened a PR from branch `y` to branch `x`, introducing a bug or whatever, but the dev knows that only when its PR has been merged to the branch `x`

Reply:
This is a good question. I limit the CI in this way to encourage developers to create PRs that only target the main branch. In this way, we limit the PRs that depend on each other. But If we think that we get more cons we can remove this limitation and trigger CI on any push 🍻

Reply:
I agree with @kin0992

Reply:
IMHO, I would run the CI even on PR with a base branch different from `main`, just to ensure that any issues can be found during the review.
I also think that our previous experiences showed us that it happens to create a PR with a base branch different from `main` (e.g. a refactor proposal).

Reply:
🍻